sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/connection/DB2ConnectionProvider.scala

```diff
@@ -48,7 +48,7 @@ private[sql] class DB2ConnectionProvider(driver: Driver, options: JDBCOptions)
     result
   }
 
-  override def setAuthenticationConfigIfNeeded(): Unit = {
+  override def setAuthenticationConfigIfNeeded(): Unit = SecurityConfigurationLock.synchronized {
```

Comment:
I've considered to make the sync on the caller side but considered bad idea because of the following:
* Not necessarily all the providers modify global JVM security config
* Maybe not the whole function must be synchronised in later providers, so this way better performing code can be implemented by synchronising the absolutely needed parts

Reply:
I don't look into the driver implementations though, is it okay to override the configuration, e.g., a  first thread sets a value at the configuration, then a second thread overrides it by a different value during the first thread scanning a JDBC relation?

Reply:
The mentioned implementation does the following:
* First checks whether there is already a config with a specific name (the name is database specific)
* If not exists then takes the actual and sets as a parent
* If exists then the later code is no-op

Worth to mention all the originally existing configuration is still reachable because if the searched application name is not matching with the newly added entry then the entry delegates the call to the parent.

I would say the operation is merge and it checks whether it's already done.

Reply:
After the details to answer the question:
> a first thread sets a value at the configuration, then a second thread overrides it by a different value

If the application name is the same then the second thread will not overwrite it but just use it.

Reply:
This behaviour is enforced by tests for each database which adds something to the security config. For instance postgres test can be found [here](https://github.com/apache/spark/blob/7735db2a273edf208ae50e88926c9f7a77e5dbac/sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/jdbc/connection/PostgresConnectionProviderSuite.scala#L20). The following code part represent the details: https://github.com/apache/spark/blob/7735db2a273edf208ae50e88926c9f7a77e5dbac/sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/jdbc/connection/ConnectionProviderSuiteBase.scala#L65-L67

Reply:
> If the application name is the same then the second thread will not overwrite it but just use it.

How about a different name case? That's what I wanna know. Since DataFrame is thread-safe, I think the case can happen.

> Worth to mention all the originally existing configuration is still reachable because if the searched application name is not matching with the newly added entry then the entry delegates the call to the parent.

If this is true, the fix looks okay. What I was worried about is a time sequence (t ~ t+3) below;

 - t: A thread-1 sets a value at the global config to scan a secure database-1. 
 - t+1: Just after the thread-1 setting, a thread-2 overrides it to scan a secure database-2 (we assume the application names are different between each other).
 - t+2: The thread-1 opens a secure connection. // <-- Can the thread-1 refer the correct own configuration?
 - t+3: The thread-2 opens it, too, then.

Reply:
Valid use-case to consider, let me explain what happens this case.

Let's assume:
* database-1 = postgres
* database-2 = db2

Execution path:
* t => security config state: `postgres` -> `empty` (JVM default)
* t+1 =>  security config state: `db2` -> `postgres` -> `empty`
* t+2 =>  thread-1 tries to reach `postgres` entry but the system config is `db2`. Since the name at `db2` entry doesn't match it forwards the request to it's parent `postgres`. The entry is matching here so this will be used for authentication.
* t+3 =>  thread-2 tries to reach `db2` entry and the system config is `db2`. The entry is matching here so this will be used for authentication.

That said the newly added security configuration is stacked together with its parent. It forwards the requests to the parent when the application name is not matching.

The writes between the threads are synchronised with the new object adde in this PR.

Reply:
> That said the newly added security configuration is stacked together with its parent.

Ah, I see. Thanks for the kind explanation. Could you write some tests for the sequence? I think its better to check if the sequence is valid in the JVM implementations.

Reply:
Sure, it's on the way...

Reply:
Just added.

Reply:
hm, is it difficult to add fine-grained tests for each connection provider in the docker-integration test side? I think the added test cannot check if secure connections work correctly, or not.

Reply:
One docker integration is is about one week, or at least that was roughly the case until now.
We have one docker integration test for each added provider at the moment.
I'm pretty sure this will change in the future because MS SQL requires active directory where I've not found any way to do it (only manual test is possible). All in all the mentioned configuration which is referred here tested one-by-one but not together so you're right.

Adding an integration test which has 2 database is definitely weeks if possible at all (not all docker images has the source code and without that hard to merge 2 DB images, at least that's my first impression). The other problem I see is that the actual integration framework is prepared to handle 1 docker image and one database connection, it would be heavy work to change that.

Reply:
hm, but I a bit hesitate to provide the new functionality that we don't have tests for. WDYT? @dongjoon-hyun @vanzin

Reply:
The functionality is already added but it contains a race bug at the moment. Let's wait on others opinion...

Reply:
To help the decision in the meantime I've tested multiple JDBC connection from a single JVM. I'm going to provide the details soon.

Reply:
So basically I've done manual integration test. Let me explain what that means. Here is my project where I've created several docker images to achieve manual kerberos connection (where possible with docker): https://github.com/gaborgsomogyi/docker-kerberos

I've build everything and then started the following docker servers:
```
./run-kdc-kadmin.sh
./run-postgres.sh
./run-mariadb.sh
```

When each and every service was up and running I've copied the `krb5.conf` and made minor modifications in the content:
```
[gaborsomogyi:~/spark] SPARK-31575-multi-test 5s ± cat /etc/krb5.conf
[libdefaults]
	udp_preference_limit=1 # This is needed to achieve TCP because Mac has problem with UDP
	default_realm = EXAMPLE.COM
	forwardable = true

[realms]
	EXAMPLE.COM = {
		kdc_ports = 88
		kadmind_port = 749
		kdc = 192.168.0.188
		admin_server = 192.168.0.188
	}
```

Added a new entry to `/etc/hosts` file:
```
192.168.0.188 example.com
```

Please note 192.168.0.188 is my local address where docker runs.

And finally I've executed the newly added `MultiKrbIntegrationSuite` test which connected to the 2 database instances within a single JVM: https://github.com/gaborgsomogyi/spark/blob/bdb573a287f26e28781084a7d8eefd10bb4ece3d/external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/MultiKrbIntegrationSuite.scala#L31

Reply:
And here is the last execution log where test succeeded:
```
Testing started at 17:57 ...
/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/bin/java "-javaagent:/Applications/IntelliJ IDEA CE.app/Contents/lib/idea_rt.jar=51444:/Applications/IntelliJ IDEA CE.app/Contents/bin" -Dfile.encoding=UTF-8 -classpath "/Users/gaborsomogyi/Library/Application Support/JetBrains/IdeaIC2020.1/plugins/Scala/lib/runners.jar:/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/jre/lib/charsets.jar:/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/jre/lib/ext/cldrdata.jar:/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/jre/lib/ext/dnsns.jar:/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/jre/lib/ext/jaccess.jar:/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/jre/lib/ext/localedata.jar:/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/jre/lib/ext/nashorn.jar:/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/jre/lib/ext/sunec.jar:/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/jre/lib/ext/sunjce_provider.jar:/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/jre/lib/ext/sunpkcs11.jar:/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/jre/lib/ext/zipfs.jar:/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/jre/lib/jce.jar:/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/jre/lib/jsse.jar:/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/jre/lib/management-agent.jar:/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/jre/lib/resources.jar:/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/jre/lib/rt.jar:/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/lib/dt.jar:/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/lib/jconsole.jar:/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/lib/sa-jdi.jar:/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/lib/tools.jar:/Users/gaborsomogyi/spark/external/docker-integration-tests/target/test-classes:/Users/gaborsomogyi/spark/external/docker-integration-tests/target/classes:/Users/gaborsomogyi/.m2/repository/com/spotify/docker-client/8.14.1/docker-client-8.14.1-shaded.jar:/Users/gaborsomogyi/.m2/repository/org/slf4j/slf4j-api/1.7.16/slf4j-api-1.7.16.jar:/Users/gaborsomogyi/.m2/repository/com/fasterxml/jackson/jaxrs/jackson-jaxrs-json-provider/2.9.6/jackson-jaxrs-json-provider-2.9.6.jar:/Users/gaborsomogyi/.m2/repository/com/fasterxml/jackson/jaxrs/jackson-jaxrs-base/2.9.6/jackson-jaxrs-base-2.9.6.jar:/Users/gaborsomogyi/.m2/repository/com/fasterxml/jackson/module/jackson-module-jaxb-annotations/2.10.0/jackson-module-jaxb-annotations-2.10.0.jar:/Users/gaborsomogyi/.m2/repository/jakarta/xml/bind/jakarta.xml.bind-api/2.3.2/jakarta.xml.bind-api-2.3.2.jar:/Users/gaborsomogyi/.m2/repository/jakarta/activation/jakarta.activation-api/1.2.1/jakarta.activation-api-1.2.1.jar:/Users/gaborsomogyi/.m2/repository/com/fasterxml/jackson/datatype/jackson-datatype-guava/2.9.6/jackson-datatype-guava-2.9.6.jar:/Users/gaborsomogyi/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.10.0/jackson-core-2.10.0.jar:/Users/gaborsomogyi/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.10.0/jackson-databind-2.10.0.jar:/Users/gaborsomogyi/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.10.0/jackson-annotations-2.10.0.jar:/Users/gaborsomogyi/.m2/repository/org/glassfish/jersey/core/jersey-client/2.30/jersey-client-2.30.jar:/Users/gaborsomogyi/.m2/repository/jakarta/ws/rs/jakarta.ws.rs-api/2.1.6/jakarta.ws.rs-api-2.1.6.jar:/Users/gaborsomogyi/.m2/repository/org/glassfish/hk2/external/jakarta.inject/2.6.1/jakarta.inject-2.6.1.jar:/Users/gaborsomogyi/.m2/repository/org/glassfish/jersey/connectors/jersey-apache-connector/2.22.2/jersey-apache-connector-2.22.2.jar:/Users/gaborsomogyi/.m2/repository/javax/ws/rs/javax.ws.rs-api/2.0.1/javax.ws.rs-api-2.0.1.jar:/Users/gaborsomogyi/.m2/repository/org/glassfish/jersey/media/jersey-media-json-jackson/2.22.2/jersey-media-json-jackson-2.22.2.jar:/Users/gaborsomogyi/.m2/repository/org/glassfish/jersey/ext/jersey-entity-filtering/2.22.2/jersey-entity-filtering-2.22.2.jar:/Users/gaborsomogyi/.m2/repository/javax/activation/activation/1.1.1/activation-1.1.1.jar:/Users/gaborsomogyi/.m2/repository/org/apache/commons/commons-compress/1.9/commons-compress-1.9.jar:/Users/gaborsomogyi/.m2/repository/commons-io/commons-io/2.4/commons-io-2.4.jar:/Users/gaborsomogyi/.m2/repository/com/github/jnr/jnr-unixsocket/0.18/jnr-unixsocket-0.18.jar:/Users/gaborsomogyi/.m2/repository/com/github/jnr/jnr-ffi/2.1.4/jnr-ffi-2.1.4.jar:/Users/gaborsomogyi/.m2/repository/com/github/jnr/jffi/1.2.15/jffi-1.2.15.jar:/Users/gaborsomogyi/.m2/repository/com/github/jnr/jffi/1.2.15/jffi-1.2.15-native.jar:/Users/gaborsomogyi/.m2/repository/org/ow2/asm/asm/5.0.3/asm-5.0.3.jar:/Users/gaborsomogyi/.m2/repository/org/ow2/asm/asm-commons/5.0.3/asm-commons-5.0.3.jar:/Users/gaborsomogyi/.m2/repository/org/ow2/asm/asm-analysis/5.0.3/asm-analysis-5.0.3.jar:/Users/gaborsomogyi/.m2/repository/org/ow2/asm/asm-tree/5.0.3/asm-tree-5.0.3.jar:/Users/gaborsomogyi/.m2/repository/org/ow2/asm/asm-util/5.0.3/asm-util-5.0.3.jar:/Users/gaborsomogyi/.m2/repository/com/github/jnr/jnr-x86asm/1.0.2/jnr-x86asm-1.0.2.jar:/Users/gaborsomogyi/.m2/repository/com/github/jnr/jnr-constants/0.9.8/jnr-constants-0.9.8.jar:/Users/gaborsomogyi/.m2/repository/com/github/jnr/jnr-enxio/0.16/jnr-enxio-0.16.jar:/Users/gaborsomogyi/.m2/repository/com/github/jnr/jnr-posix/3.0.35/jnr-posix-3.0.35.jar:/Users/gaborsomogyi/.m2/repository/commons-lang/commons-lang/2.6/commons-lang-2.6.jar:/Users/gaborsomogyi/.m2/repository/org/bouncycastle/bcpkix-jdk15on/1.59/bcpkix-jdk15on-1.59.jar:/Users/gaborsomogyi/.m2/repository/org/bouncycastle/bcprov-jdk15on/1.59/bcprov-jdk15on-1.59.jar:/Users/gaborsomogyi/.m2/repository/org/apache/httpcomponents/httpclient/4.5.6/httpclient-4.5.6.jar:/Users/gaborsomogyi/.m2/repository/commons-logging/commons-logging/1.1.3/commons-logging-1.1.3.jar:/Users/gaborsomogyi/.m2/repository/commons-codec/commons-codec/1.10/commons-codec-1.10.jar:/Users/gaborsomogyi/.m2/repository/org/apache/httpcomponents/httpcore/4.4.12/httpcore-4.4.12.jar:/Users/gaborsomogyi/.m2/repository/com/google/guava/guava/18.0/guava-18.0.jar:/Users/gaborsomogyi/spark/core/target/scala-2.12/classes:/Users/gaborsomogyi/.m2/repository/com/thoughtworks/paranamer/paranamer/2.8/paranamer-2.8.jar:/Users/gaborsomogyi/.m2/repository/org/apache/avro/avro/1.8.2/avro-1.8.2.jar:/Users/gaborsomogyi/.m2/repository/org/codehaus/jackson/jackson-core-asl/1.9.13/jackson-core-asl-1.9.13.jar:/Users/gaborsomogyi/.m2/repository/org/codehaus/jackson/jackson-mapper-asl/1.9.13/jackson-mapper-asl-1.9.13.jar:/Users/gaborsomogyi/.m2/repository/org/tukaani/xz/1.5/xz-1.5.jar:/Users/gaborsomogyi/.m2/repository/org/apache/avro/avro-mapred/1.8.2/avro-mapred-1.8.2-hadoop2.jar:/Users/gaborsomogyi/.m2/repository/org/apache/avro/avro-ipc/1.8.2/avro-ipc-1.8.2.jar:/Users/gaborsomogyi/.m2/repository/com/twitter/chill_2.12/0.9.5/chill_2.12-0.9.5.jar:/Users/gaborsomogyi/.m2/repository/com/esotericsoftware/kryo-shaded/4.0.2/kryo-shaded-4.0.2.jar:/Users/gaborsomogyi/.m2/repository/com/esotericsoftware/minlog/1.3.0/minlog-1.3.0.jar:/Users/gaborsomogyi/.m2/repository/org/objenesis/objenesis/2.5.1/objenesis-2.5.1.jar:/Users/gaborsomogyi/.m2/repository/com/twitter/chill-java/0.9.5/chill-java-0.9.5.jar:/Users/gaborsomogyi/.m2/repository/org/apache/xbean/xbean-asm7-shaded/4.15/xbean-asm7-shaded-4.15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/hadoop/hadoop-client/2.7.4/hadoop-client-2.7.4.jar:/Users/gaborsomogyi/.m2/repository/org/apache/hadoop/hadoop-common/2.7.4/hadoop-common-2.7.4.jar:/Users/gaborsomogyi/.m2/repository/commons-cli/commons-cli/1.2/commons-cli-1.2.jar:/Users/gaborsomogyi/.m2/repository/xmlenc/xmlenc/0.52/xmlenc-0.52.jar:/Users/gaborsomogyi/.m2/repository/commons-httpclient/commons-httpclient/3.1/commons-httpclient-3.1.jar:/Users/gaborsomogyi/.m2/repository/org/mortbay/jetty/jetty-sslengine/6.1.26/jetty-sslengine-6.1.26.jar:/Users/gaborsomogyi/.m2/repository/javax/servlet/jsp/jsp-api/2.1/jsp-api-2.1.jar:/Users/gaborsomogyi/.m2/repository/commons-configuration/commons-configuration/1.6/commons-configuration-1.6.jar:/Users/gaborsomogyi/.m2/repository/commons-digester/commons-digester/1.8/commons-digester-1.8.jar:/Users/gaborsomogyi/.m2/repository/commons-beanutils/commons-beanutils/1.9.4/commons-beanutils-1.9.4.jar:/Users/gaborsomogyi/.m2/repository/com/google/code/gson/gson/2.2.4/gson-2.2.4.jar:/Users/gaborsomogyi/.m2/repository/org/apache/hadoop/hadoop-auth/2.7.4/hadoop-auth-2.7.4.jar:/Users/gaborsomogyi/.m2/repository/org/apache/curator/curator-client/2.7.1/curator-client-2.7.1.jar:/Users/gaborsomogyi/.m2/repository/org/apache/htrace/htrace-core/3.1.0-incubating/htrace-core-3.1.0-incubating.jar:/Users/gaborsomogyi/.m2/repository/org/apache/hadoop/hadoop-hdfs/2.7.4/hadoop-hdfs-2.7.4.jar:/Users/gaborsomogyi/.m2/repository/org/mortbay/jetty/jetty-util/6.1.26/jetty-util-6.1.26.jar:/Users/gaborsomogyi/.m2/repository/xerces/xercesImpl/2.12.0/xercesImpl-2.12.0.jar:/Users/gaborsomogyi/.m2/repository/xml-apis/xml-apis/1.4.01/xml-apis-1.4.01.jar:/Users/gaborsomogyi/.m2/repository/org/apache/hadoop/hadoop-mapreduce-client-app/2.7.4/hadoop-mapreduce-client-app-2.7.4.jar:/Users/gaborsomogyi/.m2/repository/org/apache/hadoop/hadoop-mapreduce-client-common/2.7.4/hadoop-mapreduce-client-common-2.7.4.jar:/Users/gaborsomogyi/.m2/repository/org/apache/hadoop/hadoop-yarn-client/2.7.4/hadoop-yarn-client-2.7.4.jar:/Users/gaborsomogyi/.m2/repository/org/apache/hadoop/hadoop-yarn-server-common/2.7.4/hadoop-yarn-server-common-2.7.4.jar:/Users/gaborsomogyi/.m2/repository/org/apache/hadoop/hadoop-mapreduce-client-shuffle/2.7.4/hadoop-mapreduce-client-shuffle-2.7.4.jar:/Users/gaborsomogyi/.m2/repository/org/apache/hadoop/hadoop-yarn-api/2.7.4/hadoop-yarn-api-2.7.4.jar:/Users/gaborsomogyi/.m2/repository/org/apache/hadoop/hadoop-mapreduce-client-core/2.7.4/hadoop-mapreduce-client-core-2.7.4.jar:/Users/gaborsomogyi/.m2/repository/org/apache/hadoop/hadoop-yarn-common/2.7.4/hadoop-yarn-common-2.7.4.jar:/Users/gaborsomogyi/.m2/repository/javax/xml/bind/jaxb-api/2.2.2/jaxb-api-2.2.2.jar:/Users/gaborsomogyi/.m2/repository/javax/xml/stream/stax-api/1.0-2/stax-api-1.0-2.jar:/Users/gaborsomogyi/.m2/repository/org/codehaus/jackson/jackson-jaxrs/1.9.13/jackson-jaxrs-1.9.13.jar:/Users/gaborsomogyi/.m2/repository/org/codehaus/jackson/jackson-xc/1.9.13/jackson-xc-1.9.13.jar:/Users/gaborsomogyi/.m2/repository/org/apache/hadoop/hadoop-mapreduce-client-jobclient/2.7.4/hadoop-mapreduce-client-jobclient-2.7.4.jar:/Users/gaborsomogyi/.m2/repository/org/apache/hadoop/hadoop-annotations/2.7.4/hadoop-annotations-2.7.4.jar:/Users/gaborsomogyi/spark/launcher/target/scala-2.12/classes:/Users/gaborsomogyi/spark/common/kvstore/target/scala-2.12/classes:/Users/gaborsomogyi/.m2/repository/org/fusesource/leveldbjni/leveldbjni-all/1.8/leveldbjni-all-1.8.jar:/Users/gaborsomogyi/spark/common/network-common/target/scala-2.12/classes:/Users/gaborsomogyi/spark/common/network-shuffle/target/scala-2.12/classes:/Users/gaborsomogyi/spark/common/unsafe/target/scala-2.12/classes:/Users/gaborsomogyi/.m2/repository/org/apache/curator/curator-recipes/2.7.1/curator-recipes-2.7.1.jar:/Users/gaborsomogyi/.m2/repository/org/apache/curator/curator-framework/2.7.1/curator-framework-2.7.1.jar:/Users/gaborsomogyi/.m2/repository/org/apache/zookeeper/zookeeper/3.4.14/zookeeper-3.4.14.jar:/Users/gaborsomogyi/.m2/repository/org/apache/yetus/audience-annotations/0.5.0/audience-annotations-0.5.0.jar:/Users/gaborsomogyi/.m2/repository/org/eclipse/jetty/jetty-plus/9.4.18.v20190429/jetty-plus-9.4.18.v20190429.jar:/Users/gaborsomogyi/.m2/repository/org/eclipse/jetty/jetty-webapp/9.4.18.v20190429/jetty-webapp-9.4.18.v20190429.jar:/Users/gaborsomogyi/.m2/repository/org/eclipse/jetty/jetty-xml/9.4.18.v20190429/jetty-xml-9.4.18.v20190429.jar:/Users/gaborsomogyi/.m2/repository/org/eclipse/jetty/jetty-jndi/9.4.18.v20190429/jetty-jndi-9.4.18.v20190429.jar:/Users/gaborsomogyi/.m2/repository/org/eclipse/jetty/jetty-security/9.4.18.v20190429/jetty-security-9.4.18.v20190429.jar:/Users/gaborsomogyi/.m2/repository/org/eclipse/jetty/jetty-util/9.4.18.v20190429/jetty-util-9.4.18.v20190429.jar:/Users/gaborsomogyi/.m2/repository/org/eclipse/jetty/jetty-server/9.4.18.v20190429/jetty-server-9.4.18.v20190429.jar:/Users/gaborsomogyi/.m2/repository/org/eclipse/jetty/jetty-io/9.4.18.v20190429/jetty-io-9.4.18.v20190429.jar:/Users/gaborsomogyi/.m2/repository/org/eclipse/jetty/jetty-http/9.4.18.v20190429/jetty-http-9.4.18.v20190429.jar:/Users/gaborsomogyi/.m2/repository/org/eclipse/jetty/jetty-continuation/9.4.18.v20190429/jetty-continuation-9.4.18.v20190429.jar:/Users/gaborsomogyi/.m2/repository/org/eclipse/jetty/jetty-servlet/9.4.18.v20190429/jetty-servlet-9.4.18.v20190429.jar:/Users/gaborsomogyi/.m2/repository/org/eclipse/jetty/jetty-proxy/9.4.18.v20190429/jetty-proxy-9.4.18.v20190429.jar:/Users/gaborsomogyi/.m2/repository/org/eclipse/jetty/jetty-client/9.4.18.v20190429/jetty-client-9.4.18.v20190429.jar:/Users/gaborsomogyi/.m2/repository/org/eclipse/jetty/jetty-servlets/9.4.18.v20190429/jetty-servlets-9.4.18.v20190429.jar:/Users/gaborsomogyi/.m2/repository/javax/servlet/javax.servlet-api/3.1.0/javax.servlet-api-3.1.0.jar:/Users/gaborsomogyi/.m2/repository/org/apache/commons/commons-lang3/3.9/commons-lang3-3.9.jar:/Users/gaborsomogyi/.m2/repository/org/apache/commons/commons-math3/3.4.1/commons-math3-3.4.1.jar:/Users/gaborsomogyi/.m2/repository/org/apache/commons/commons-text/1.6/commons-text-1.6.jar:/Users/gaborsomogyi/.m2/repository/com/google/code/findbugs/jsr305/3.0.0/jsr305-3.0.0.jar:/Users/gaborsomogyi/.m2/repository/org/slf4j/jul-to-slf4j/1.7.16/jul-to-slf4j-1.7.16.jar:/Users/gaborsomogyi/.m2/repository/org/slf4j/jcl-over-slf4j/1.7.16/jcl-over-slf4j-1.7.16.jar:/Users/gaborsomogyi/.m2/repository/log4j/log4j/1.2.17/log4j-1.2.17.jar:/Users/gaborsomogyi/.m2/repository/org/slf4j/slf4j-log4j12/1.7.16/slf4j-log4j12-1.7.16.jar:/Users/gaborsomogyi/.m2/repository/com/ning/compress-lzf/1.0.3/compress-lzf-1.0.3.jar:/Users/gaborsomogyi/.m2/repository/org/xerial/snappy/snappy-java/1.1.7.3/snappy-java-1.1.7.3.jar:/Users/gaborsomogyi/.m2/repository/org/lz4/lz4-java/1.7.1/lz4-java-1.7.1.jar:/Users/gaborsomogyi/.m2/repository/com/github/luben/zstd-jni/1.4.4-3/zstd-jni-1.4.4-3.jar:/Users/gaborsomogyi/.m2/repository/org/roaringbitmap/RoaringBitmap/0.7.45/RoaringBitmap-0.7.45.jar:/Users/gaborsomogyi/.m2/repository/org/roaringbitmap/shims/0.7.45/shims-0.7.45.jar:/Users/gaborsomogyi/.m2/repository/commons-net/commons-net/3.1/commons-net-3.1.jar:/Users/gaborsomogyi/.m2/repository/org/scala-lang/modules/scala-xml_2.12/1.2.0/scala-xml_2.12-1.2.0.jar:/Users/gaborsomogyi/.m2/repository/org/scala-lang/scala-library/2.12.10/scala-library-2.12.10.jar:/Users/gaborsomogyi/.m2/repository/org/scala-lang/scala-reflect/2.12.10/scala-reflect-2.12.10.jar:/Users/gaborsomogyi/.m2/repository/org/json4s/json4s-jackson_2.12/3.6.6/json4s-jackson_2.12-3.6.6.jar:/Users/gaborsomogyi/.m2/repository/org/json4s/json4s-core_2.12/3.6.6/json4s-core_2.12-3.6.6.jar:/Users/gaborsomogyi/.m2/repository/org/json4s/json4s-ast_2.12/3.6.6/json4s-ast_2.12-3.6.6.jar:/Users/gaborsomogyi/.m2/repository/org/json4s/json4s-scalap_2.12/3.6.6/json4s-scalap_2.12-3.6.6.jar:/Users/gaborsomogyi/.m2/repository/org/glassfish/jersey/core/jersey-common/2.30/jersey-common-2.30.jar:/Users/gaborsomogyi/.m2/repository/jakarta/annotation/jakarta.annotation-api/1.3.5/jakarta.annotation-api-1.3.5.jar:/Users/gaborsomogyi/.m2/repository/org/glassfish/hk2/osgi-resource-locator/1.0.3/osgi-resource-locator-1.0.3.jar:/Users/gaborsomogyi/.m2/repository/org/glassfish/jersey/core/jersey-server/2.30/jersey-server-2.30.jar:/Users/gaborsomogyi/.m2/repository/org/glassfish/jersey/media/jersey-media-jaxb/2.30/jersey-media-jaxb-2.30.jar:/Users/gaborsomogyi/.m2/repository/jakarta/validation/jakarta.validation-api/2.0.2/jakarta.validation-api-2.0.2.jar:/Users/gaborsomogyi/.m2/repository/org/glassfish/jersey/containers/jersey-container-servlet/2.30/jersey-container-servlet-2.30.jar:/Users/gaborsomogyi/.m2/repository/org/glassfish/jersey/containers/jersey-container-servlet-core/2.30/jersey-container-servlet-core-2.30.jar:/Users/gaborsomogyi/.m2/repository/org/glassfish/jersey/inject/jersey-hk2/2.30/jersey-hk2-2.30.jar:/Users/gaborsomogyi/.m2/repository/org/glassfish/hk2/hk2-locator/2.6.1/hk2-locator-2.6.1.jar:/Users/gaborsomogyi/.m2/repository/org/glassfish/hk2/external/aopalliance-repackaged/2.6.1/aopalliance-repackaged-2.6.1.jar:/Users/gaborsomogyi/.m2/repository/org/glassfish/hk2/hk2-api/2.6.1/hk2-api-2.6.1.jar:/Users/gaborsomogyi/.m2/repository/org/glassfish/hk2/hk2-utils/2.6.1/hk2-utils-2.6.1.jar:/Users/gaborsomogyi/.m2/repository/org/javassist/javassist/3.25.0-GA/javassist-3.25.0-GA.jar:/Users/gaborsomogyi/.m2/repository/io/netty/netty-all/4.1.47.Final/netty-all-4.1.47.Final.jar:/Users/gaborsomogyi/.m2/repository/com/clearspring/analytics/stream/2.9.6/stream-2.9.6.jar:/Users/gaborsomogyi/.m2/repository/io/dropwizard/metrics/metrics-core/4.1.1/metrics-core-4.1.1.jar:/Users/gaborsomogyi/.m2/repository/io/dropwizard/metrics/metrics-jvm/4.1.1/metrics-jvm-4.1.1.jar:/Users/gaborsomogyi/.m2/repository/io/dropwizard/metrics/metrics-json/4.1.1/metrics-json-4.1.1.jar:/Users/gaborsomogyi/.m2/repository/io/dropwizard/metrics/metrics-graphite/4.1.1/metrics-graphite-4.1.1.jar:/Users/gaborsomogyi/.m2/repository/io/dropwizard/metrics/metrics-jmx/4.1.1/metrics-jmx-4.1.1.jar:/Users/gaborsomogyi/.m2/repository/com/fasterxml/jackson/module/jackson-module-scala_2.12/2.10.0/jackson-module-scala_2.12-2.10.0.jar:/Users/gaborsomogyi/.m2/repository/com/fasterxml/jackson/module/jackson-module-paranamer/2.10.0/jackson-module-paranamer-2.10.0.jar:/Users/gaborsomogyi/.m2/repository/org/apache/ivy/ivy/2.4.0/ivy-2.4.0.jar:/Users/gaborsomogyi/.m2/repository/oro/oro/2.0.8/oro-2.0.8.jar:/Users/gaborsomogyi/.m2/repository/net/razorvine/pyrolite/4.30/pyrolite-4.30.jar:/Users/gaborsomogyi/.m2/repository/net/sf/py4j/py4j/0.10.9/py4j-0.10.9.jar:/Users/gaborsomogyi/spark/common/tags/target/scala-2.12/classes:/Users/gaborsomogyi/.m2/repository/org/apache/commons/commons-crypto/1.0.0/commons-crypto-1.0.0.jar:/Users/gaborsomogyi/spark/core/target/scala-2.12/test-classes:/Users/gaborsomogyi/spark/sql/catalyst/target/scala-2.12/test-classes:/Users/gaborsomogyi/.m2/repository/org/scala-lang/modules/scala-parser-combinators_2.12/1.1.2/scala-parser-combinators_2.12-1.1.2.jar:/Users/gaborsomogyi/spark/common/sketch/target/scala-2.12/classes:/Users/gaborsomogyi/.m2/repository/org/codehaus/janino/janino/3.0.16/janino-3.0.16.jar:/Users/gaborsomogyi/.m2/repository/org/codehaus/janino/commons-compiler/3.0.16/commons-compiler-3.0.16.jar:/Users/gaborsomogyi/.m2/repository/org/antlr/antlr4-runtime/4.7.1/antlr4-runtime-4.7.1.jar:/Users/gaborsomogyi/.m2/repository/com/univocity/univocity-parsers/2.8.3/univocity-parsers-2.8.3.jar:/Users/gaborsomogyi/.m2/repository/org/apache/arrow/arrow-vector/0.15.1/arrow-vector-0.15.1.jar:/Users/gaborsomogyi/.m2/repository/org/apache/arrow/arrow-format/0.15.1/arrow-format-0.15.1.jar:/Users/gaborsomogyi/.m2/repository/org/apache/arrow/arrow-memory/0.15.1/arrow-memory-0.15.1.jar:/Users/gaborsomogyi/.m2/repository/com/google/flatbuffers/flatbuffers-java/1.9.0/flatbuffers-java-1.9.0.jar:/Users/gaborsomogyi/spark/sql/core/target/scala-2.12/classes:/Users/gaborsomogyi/spark/sql/catalyst/target/scala-2.12/classes:/Users/gaborsomogyi/.m2/repository/org/apache/orc/orc-core/1.5.9/orc-core-1.5.9.jar:/Users/gaborsomogyi/.m2/repository/org/apache/orc/orc-shims/1.5.9/orc-shims-1.5.9.jar:/Users/gaborsomogyi/.m2/repository/com/google/protobuf/protobuf-java/2.5.0/protobuf-java-2.5.0.jar:/Users/gaborsomogyi/.m2/repository/io/airlift/aircompressor/0.10/aircompressor-0.10.jar:/Users/gaborsomogyi/.m2/repository/org/threeten/threeten-extra/1.5.0/threeten-extra-1.5.0.jar:/Users/gaborsomogyi/.m2/repository/org/apache/orc/orc-mapreduce/1.5.9/orc-mapreduce-1.5.9.jar:/Users/gaborsomogyi/.m2/repository/org/apache/hive/hive-storage-api/2.7.1/hive-storage-api-2.7.1.jar:/Users/gaborsomogyi/.m2/repository/org/apache/parquet/parquet-column/1.10.1/parquet-column-1.10.1.jar:/Users/gaborsomogyi/.m2/repository/org/apache/parquet/parquet-common/1.10.1/parquet-common-1.10.1.jar:/Users/gaborsomogyi/.m2/repository/org/apache/parquet/parquet-encoding/1.10.1/parquet-encoding-1.10.1.jar:/Users/gaborsomogyi/.m2/repository/org/apache/parquet/parquet-hadoop/1.10.1/parquet-hadoop-1.10.1.jar:/Users/gaborsomogyi/.m2/repository/org/apache/parquet/parquet-format/2.4.0/parquet-format-2.4.0.jar:/Users/gaborsomogyi/.m2/repository/org/apache/parquet/parquet-jackson/1.10.1/parquet-jackson-1.10.1.jar:/Users/gaborsomogyi/spark/sql/core/target/scala-2.12/test-classes:/Users/gaborsomogyi/spark/common/tags/target/scala-2.12/test-classes:/Users/gaborsomogyi/.m2/repository/org/apache/hadoop/hadoop-minikdc/2.7.4/hadoop-minikdc-2.7.4.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-core-api/2.0.0-M15/apacheds-core-api-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-core-constants/2.0.0-M15/apacheds-core-constants-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-i18n/2.0.0-M15/apacheds-i18n-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/api/api-i18n/1.0.0-M20/api-i18n-1.0.0-M20.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/api/api-asn1-api/1.0.0-M20/api-asn1-api-1.0.0-M20.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/api/api-ldap-client-api/1.0.0-M20/api-ldap-client-api-1.0.0-M20.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/api/api-ldap-codec-core/1.0.0-M20/api-ldap-codec-core-1.0.0-M20.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/api/api-ldap-extras-aci/1.0.0-M20/api-ldap-extras-aci-1.0.0-M20.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/api/api-ldap-extras-util/1.0.0-M20/api-ldap-extras-util-1.0.0-M20.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/api/api-ldap-model/1.0.0-M20/api-ldap-model-1.0.0-M20.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/api/api-util/1.0.0-M20/api-util-1.0.0-M20.jar:/Users/gaborsomogyi/.m2/repository/org/apache/mina/mina-core/2.0.7/mina-core-2.0.7.jar:/Users/gaborsomogyi/.m2/repository/net/sf/ehcache/ehcache-core/2.4.4/ehcache-core-2.4.4.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-interceptor-kerberos/2.0.0-M15/apacheds-interceptor-kerberos-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-core/2.0.0-M15/apacheds-core-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-interceptors-admin/2.0.0-M15/apacheds-interceptors-admin-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-interceptors-authn/2.0.0-M15/apacheds-interceptors-authn-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-interceptors-authz/2.0.0-M15/apacheds-interceptors-authz-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-interceptors-changelog/2.0.0-M15/apacheds-interceptors-changelog-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-interceptors-collective/2.0.0-M15/apacheds-interceptors-collective-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-interceptors-event/2.0.0-M15/apacheds-interceptors-event-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-interceptors-exception/2.0.0-M15/apacheds-interceptors-exception-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-interceptors-journal/2.0.0-M15/apacheds-interceptors-journal-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-interceptors-normalization/2.0.0-M15/apacheds-interceptors-normalization-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-interceptors-operational/2.0.0-M15/apacheds-interceptors-operational-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-interceptors-referral/2.0.0-M15/apacheds-interceptors-referral-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-interceptors-schema/2.0.0-M15/apacheds-interceptors-schema-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-interceptors-subtree/2.0.0-M15/apacheds-interceptors-subtree-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-interceptors-trigger/2.0.0-M15/apacheds-interceptors-trigger-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/api/api-ldap-extras-trigger/1.0.0-M20/api-ldap-extras-trigger-1.0.0-M20.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-kerberos-codec/2.0.0-M15/apacheds-kerberos-codec-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-protocol-shared/2.0.0-M15/apacheds-protocol-shared-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-protocol-kerberos/2.0.0-M15/apacheds-protocol-kerberos-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-core-shared/2.0.0-M15/apacheds-core-shared-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-ldif-partition/2.0.0-M15/apacheds-ldif-partition-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-xdbm-partition/2.0.0-M15/apacheds-xdbm-partition-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-mavibot-partition/2.0.0-M15/apacheds-mavibot-partition-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/mavibot/mavibot/1.0.0-M1/mavibot-1.0.0-M1.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-core-avl/2.0.0-M15/apacheds-core-avl-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/api/api-all/1.0.0-M20/api-all-1.0.0-M20.jar:/Users/gaborsomogyi/.m2/repository/commons-pool/commons-pool/1.6/commons-pool-1.6.jar:/Users/gaborsomogyi/.m2/repository/commons-collections/commons-collections/3.2.2/commons-collections-3.2.2.jar:/Users/gaborsomogyi/.m2/repository/antlr/antlr/2.7.7/antlr-2.7.7.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-jdbm-partition/2.0.0-M15/apacheds-jdbm-partition-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/jdbm/apacheds-jdbm1/2.0.0-M2/apacheds-jdbm1-2.0.0-M2.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/server/apacheds-protocol-ldap/2.0.0-M15/apacheds-protocol-ldap-2.0.0-M15.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/api/api-asn1-ber/1.0.0-M20/api-asn1-ber-1.0.0-M20.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/api/api-ldap-extras-codec-api/1.0.0-M20/api-ldap-extras-codec-api-1.0.0-M20.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/api/api-ldap-extras-codec/1.0.0-M20/api-ldap-extras-codec-1.0.0-M20.jar:/Users/gaborsomogyi/.m2/repository/org/apache/directory/api/api-ldap-extras-sp/1.0.0-M20/api-ldap-extras-sp-1.0.0-M20.jar:/Users/gaborsomogyi/.m2/repository/bouncycastle/bcprov-jdk15/140/bcprov-jdk15-140.jar:/Users/gaborsomogyi/.m2/repository/org/glassfish/jersey/bundles/repackaged/jersey-guava/2.25.1/jersey-guava-2.25.1.jar:/Users/gaborsomogyi/.m2/repository/org/mariadb/jdbc/mariadb-java-client/2.5.4/mariadb-java-client-2.5.4.jar:/Users/gaborsomogyi/.m2/repository/org/postgresql/postgresql/42.2.6/postgresql-42.2.6.jar:/Users/gaborsomogyi/.m2/repository/com/oracle/ojdbc6/11.2.0.1.0/ojdbc6-11.2.0.1.0.jar:/Users/gaborsomogyi/.m2/repository/com/ibm/db2/jcc/11.5.0.0/jcc-11.5.0.0.jar:/Users/gaborsomogyi/.m2/repository/com/microsoft/sqlserver/mssql-jdbc/7.2.1.jre8/mssql-jdbc-7.2.1.jre8.jar:/Users/gaborsomogyi/.m2/repository/org/spark-project/spark/unused/1.0.0/unused-1.0.0.jar:/Users/gaborsomogyi/.m2/repository/org/scalatest/scalatest_2.12/3.0.8/scalatest_2.12-3.0.8.jar:/Users/gaborsomogyi/.m2/repository/org/scalactic/scalactic_2.12/3.0.8/scalactic_2.12-3.0.8.jar:/Users/gaborsomogyi/.m2/repository/junit/junit/4.12/junit-4.12.jar:/Users/gaborsomogyi/.m2/repository/org/hamcrest/hamcrest-core/1.3/hamcrest-core-1.3.jar:/Users/gaborsomogyi/.m2/repository/com/novocode/junit-interface/0.11/junit-interface-0.11.jar:/Users/gaborsomogyi/.m2/repository/org/scala-sbt/test-interface/1.0/test-interface-1.0.jar" org.jetbrains.plugins.scala.testingSupport.scalaTest.ScalaTestRunner -s org.apache.spark.sql.jdbc.MultiKrbIntegrationSuite -C org.jetbrains.plugins.scala.testingSupport.scalaTest.ScalaTestReporter -showProgressMessages true


20/05/05 17:57:40.697 ScalaTest-run INFO SparkContext: Running Spark version 3.1.0-SNAPSHOT
20/05/05 17:57:41.005 ScalaTest-run WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
20/05/05 17:57:41.081 ScalaTest-run INFO ResourceUtils: ==============================================================
20/05/05 17:57:41.082 ScalaTest-run INFO ResourceUtils: No custom resources configured for spark.driver.
20/05/05 17:57:41.082 ScalaTest-run INFO ResourceUtils: ==============================================================
20/05/05 17:57:41.082 ScalaTest-run INFO SparkContext: Submitted application: test-sql-context
20/05/05 17:57:41.098 ScalaTest-run INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(cores -> name: cores, amount: 1, script: , vendor: , memory -> name: memory, amount: 1024, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
20/05/05 17:57:41.109 ScalaTest-run INFO ResourceProfile: Limiting resource is cpu
20/05/05 17:57:41.109 ScalaTest-run INFO ResourceProfileManager: Added ResourceProfile id: 0
20/05/05 17:57:41.156 ScalaTest-run INFO SecurityManager: Changing view acls to: gaborsomogyi
20/05/05 17:57:41.157 ScalaTest-run INFO SecurityManager: Changing modify acls to: gaborsomogyi
20/05/05 17:57:41.157 ScalaTest-run INFO SecurityManager: Changing view acls groups to: 
20/05/05 17:57:41.158 ScalaTest-run INFO SecurityManager: Changing modify acls groups to: 
20/05/05 17:57:41.158 ScalaTest-run INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users  with view permissions: Set(gaborsomogyi); groups with view permissions: Set(); users  with modify permissions: Set(gaborsomogyi); groups with modify permissions: Set()
20/05/05 17:57:41.438 ScalaTest-run INFO Utils: Successfully started service 'sparkDriver' on port 51448.
20/05/05 17:57:41.462 ScalaTest-run INFO SparkEnv: Registering MapOutputTracker
20/05/05 17:57:41.487 ScalaTest-run INFO SparkEnv: Registering BlockManagerMaster
20/05/05 17:57:41.490 ScalaTest-run INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
20/05/05 17:57:41.490 ScalaTest-run INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
20/05/05 17:57:41.493 ScalaTest-run INFO SparkEnv: Registering BlockManagerMasterHeartbeat
20/05/05 17:57:41.513 ScalaTest-run INFO DiskBlockManager: Created local directory at /private/var/folders/jd/35_sh46s7zq0qc6khfw8hc800000gn/T/blockmgr-394e7546-1c3c-49c1-b872-9d7e997082de
20/05/05 17:57:41.542 ScalaTest-run INFO MemoryStore: MemoryStore started with capacity 4.1 GiB
20/05/05 17:57:41.558 ScalaTest-run INFO SparkEnv: Registering OutputCommitCoordinator
20/05/05 17:57:41.637 ScalaTest-run INFO log: Logging initialized @3323ms to org.eclipse.jetty.util.log.Slf4jLog
20/05/05 17:57:41.723 ScalaTest-run INFO Server: jetty-9.4.18.v20190429; built: 2019-04-29T20:42:08.989Z; git: e1bc35120a6617ee3df052294e433f3a25ce7097; jvm 1.8.0_242-b08
20/05/05 17:57:41.751 ScalaTest-run INFO Server: Started @3438ms
20/05/05 17:57:41.786 ScalaTest-run INFO AbstractConnector: Started ServerConnector@4985cbcb{HTTP/1.1,[http/1.1]}{0.0.0.0:4040}
20/05/05 17:57:41.786 ScalaTest-run INFO Utils: Successfully started service 'SparkUI' on port 4040.
20/05/05 17:57:41.838 ScalaTest-run INFO ContextHandler: Started o.e.j.s.ServletContextHandler@7f811d00{/jobs,null,AVAILABLE,@Spark}
20/05/05 17:57:41.839 ScalaTest-run INFO ContextHandler: Started o.e.j.s.ServletContextHandler@3bffddff{/jobs/json,null,AVAILABLE,@Spark}
20/05/05 17:57:41.840 ScalaTest-run INFO ContextHandler: Started o.e.j.s.ServletContextHandler@50687efb{/jobs/job,null,AVAILABLE,@Spark}
20/05/05 17:57:41.841 ScalaTest-run INFO ContextHandler: Started o.e.j.s.ServletContextHandler@4a9cc6cb{/jobs/job/json,null,AVAILABLE,@Spark}
20/05/05 17:57:41.841 ScalaTest-run INFO ContextHandler: Started o.e.j.s.ServletContextHandler@56e07a08{/stages,null,AVAILABLE,@Spark}
20/05/05 17:57:41.842 ScalaTest-run INFO ContextHandler: Started o.e.j.s.ServletContextHandler@1950e8a6{/stages/json,null,AVAILABLE,@Spark}
20/05/05 17:57:41.842 ScalaTest-run INFO ContextHandler: Started o.e.j.s.ServletContextHandler@12cd9150{/stages/stage,null,AVAILABLE,@Spark}
20/05/05 17:57:41.843 ScalaTest-run INFO ContextHandler: Started o.e.j.s.ServletContextHandler@724f138e{/stages/stage/json,null,AVAILABLE,@Spark}
20/05/05 17:57:41.844 ScalaTest-run INFO ContextHandler: Started o.e.j.s.ServletContextHandler@32fe9d0a{/stages/pool,null,AVAILABLE,@Spark}
20/05/05 17:57:41.844 ScalaTest-run INFO ContextHandler: Started o.e.j.s.ServletContextHandler@64da2a7{/stages/pool/json,null,AVAILABLE,@Spark}
20/05/05 17:57:41.845 ScalaTest-run INFO ContextHandler: Started o.e.j.s.ServletContextHandler@d78795{/storage,null,AVAILABLE,@Spark}
20/05/05 17:57:41.845 ScalaTest-run INFO ContextHandler: Started o.e.j.s.ServletContextHandler@47428937{/storage/json,null,AVAILABLE,@Spark}
20/05/05 17:57:41.846 ScalaTest-run INFO ContextHandler: Started o.e.j.s.ServletContextHandler@7caa550{/storage/rdd,null,AVAILABLE,@Spark}
20/05/05 17:57:41.846 ScalaTest-run INFO ContextHandler: Started o.e.j.s.ServletContextHandler@72b16078{/storage/rdd/json,null,AVAILABLE,@Spark}
20/05/05 17:57:41.847 ScalaTest-run INFO ContextHandler: Started o.e.j.s.ServletContextHandler@70fab835{/environment,null,AVAILABLE,@Spark}
20/05/05 17:57:41.847 ScalaTest-run INFO ContextHandler: Started o.e.j.s.ServletContextHandler@62417a16{/environment/json,null,AVAILABLE,@Spark}
20/05/05 17:57:41.848 ScalaTest-run INFO ContextHandler: Started o.e.j.s.ServletContextHandler@26be6ca7{/executors,null,AVAILABLE,@Spark}
20/05/05 17:57:41.848 ScalaTest-run INFO ContextHandler: Started o.e.j.s.ServletContextHandler@759fad4{/executors/json,null,AVAILABLE,@Spark}
20/05/05 17:57:41.849 ScalaTest-run INFO ContextHandler: Started o.e.j.s.ServletContextHandler@53499d85{/executors/threadDump,null,AVAILABLE,@Spark}
20/05/05 17:57:41.849 ScalaTest-run INFO ContextHandler: Started o.e.j.s.ServletContextHandler@782a4fff{/executors/threadDump/json,null,AVAILABLE,@Spark}
20/05/05 17:57:41.861 ScalaTest-run INFO ContextHandler: Started o.e.j.s.ServletContextHandler@59fc684e{/static,null,AVAILABLE,@Spark}
20/05/05 17:57:41.861 ScalaTest-run INFO ContextHandler: Started o.e.j.s.ServletContextHandler@363f6148{/,null,AVAILABLE,@Spark}
20/05/05 17:57:41.862 ScalaTest-run INFO ContextHandler: Started o.e.j.s.ServletContextHandler@1b28f282{/api,null,AVAILABLE,@Spark}
20/05/05 17:57:41.862 ScalaTest-run INFO ContextHandler: Started o.e.j.s.ServletContextHandler@29876704{/jobs/job/kill,null,AVAILABLE,@Spark}
20/05/05 17:57:41.863 ScalaTest-run INFO ContextHandler: Started o.e.j.s.ServletContextHandler@4940809c{/stages/stage/kill,null,AVAILABLE,@Spark}
20/05/05 17:57:41.864 ScalaTest-run INFO SparkUI: Bound SparkUI to 0.0.0.0, and started at http://example.com:4040
20/05/05 17:57:41.999 ScalaTest-run INFO Executor: Starting executor ID driver on host example.com
20/05/05 17:57:42.034 ScalaTest-run INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 51449.
20/05/05 17:57:42.035 ScalaTest-run INFO NettyBlockTransferService: Server created on example.com:51449
20/05/05 17:57:42.039 ScalaTest-run INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
20/05/05 17:57:42.046 ScalaTest-run INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, example.com, 51449, None)
20/05/05 17:57:42.050 dispatcher-BlockManagerMaster INFO BlockManagerMasterEndpoint: Registering block manager example.com:51449 with 4.1 GiB RAM, BlockManagerId(driver, example.com, 51449, None)
20/05/05 17:57:42.053 ScalaTest-run INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, example.com, 51449, None)
20/05/05 17:57:42.057 ScalaTest-run INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, example.com, 51449, None)
20/05/05 17:57:42.178 ScalaTest-run INFO ContextHandler: Started o.e.j.s.ServletContextHandler@435ce306{/metrics/json,null,AVAILABLE,@Spark}

20/05/05 17:57:42.253 ScalaTest-run-running-MultiKrbIntegrationSuite INFO MultiKrbIntegrationSuite: 

===== TEST OUTPUT FOR o.a.s.sql.jdbc.MultiKrbIntegrationSuite: 'Basic multi test' =====

Debug is  true storeKey false useTicketCache false useKeyTab true doNotPrompt false ticketCache is null isInitiator true KeyTab is /Users/gaborsomogyi/docker-kerberos/postgres_share/postgres.keytab refreshKrb5Config is false principal is postgres/example.com@EXAMPLE.COM tryFirstPass is false useFirstPass is false storePass is false clearPass is false
principal is postgres/example.com@EXAMPLE.COM
Will use keytab
Commit Succeeded 

Found ticket for postgres/example.com@EXAMPLE.COM to go to krbtgt/EXAMPLE.COM@EXAMPLE.COM expiring on Wed May 06 17:57:42 CEST 2020
Entered Krb5Context.initSecContext with state=STATE_NEW
Found ticket for postgres/example.com@EXAMPLE.COM to go to krbtgt/EXAMPLE.COM@EXAMPLE.COM expiring on Wed May 06 17:57:42 CEST 2020
Service ticket not found in the subject
Krb5Context setting mySeqNumber to: 386497368
Created InitSecContextToken:
0000: 01 00 6E 82 02 4F 30 82   02 4B A0 03 02 01 05 A1  ..n..O0..K......
0010: 03 02 01 0E A2 07 03 05   00 20 00 00 00 A3 82 01  ......... ......
0020: 53 61 82 01 4F 30 82 01   4B A0 03 02 01 05 A1 0D  Sa..O0..K.......
0030: 1B 0B 45 58 41 4D 50 4C   45 2E 43 4F 4D A2 22 30  ..EXAMPLE.COM."0
0040: 20 A0 03 02 01 03 A1 19   30 17 1B 08 70 6F 73 74   .......0...post
0050: 67 72 65 73 1B 0B 65 78   61 6D 70 6C 65 2E 63 6F  gres..example.co
0060: 6D A3 82 01 0F 30 82 01   0B A0 03 02 01 12 A1 03  m....0..........
0070: 02 01 02 A2 81 FE 04 81   FB 9E 9B 9B 30 B8 03 11  ............0...
0080: 4D 21 51 B6 35 DD D9 D7   89 CF E8 F1 ED B2 4A 98  M!Q.5.........J.
0090: FB 19 58 09 49 E8 16 ED   88 E9 72 12 4B DA 96 70  ..X.I.....r.K..p
00A0: FC BA D1 50 7E A0 E8 15   72 4E 06 C4 0E D8 0F 97  ...P....rN......
00B0: 7A D4 39 9F E2 42 59 53   FF 48 67 42 BC C1 06 30  z.9..BYS.HgB...0
00C0: E1 45 D0 0B 19 94 FC F2   C6 5F B1 16 A3 F0 A7 90  .E......._......
00D0: D4 AD C5 FC 25 57 DE 15   AD EB AC AA 9F 8D 63 BF  ....%W........c.
00E0: A0 90 4D 37 E1 BA 0E 7D   94 C8 55 8C 3A 99 73 1D  ..M7......U.:.s.
00F0: C2 A1 0C 55 6F 6F 4E EE   80 1B 19 92 82 60 85 51  ...UooN......`.Q
0100: A9 E8 7A CC 6E DC 14 DD   4F B8 AB 2D 2B EC 81 C2  ..z.n...O..-+...
0110: 33 39 E0 DE F2 D5 4E 2C   92 5F 79 3A 30 B8 FD 78  39....N,._y:0..x
0120: 69 81 8B 1C D4 0C E4 10   5C 79 C0 26 4A 89 A8 1F  i.......\y.&J...
0130: D3 1B B9 D8 57 47 AB 40   61 B8 C9 52 CC 73 96 E2  ....WG.@a..R.s..
0140: F7 FE D1 8E DF 3B 62 70   63 C4 5E CF AE A4 EF 99  .....;bpc.^.....
0150: 40 6A 82 8A E6 5C 92 F0   1E A1 8D 51 06 49 B9 1D  @j...\.....Q.I..
0160: 99 FC 1E B3 56 F5 B3 E7   82 B8 49 56 9E E8 90 E1  ....V.....IV....
0170: F6 F9 BC 05 A4 81 DE 30   81 DB A0 03 02 01 12 A2  .......0........
0180: 81 D3 04 81 D0 95 D6 66   21 D8 E7 EC C2 E9 F1 6D  .......f!......m
0190: 55 CA 2A A3 80 3A 5C 0D   2B 5A 8E C7 01 68 5E 02  U.*..:\.+Z...h^.
01A0: FB 72 24 C2 72 51 E1 59   56 23 14 3E 42 90 6E AE  .r$.rQ.YV#.>B.n.
01B0: 1F 33 D9 B1 43 E7 27 7C   71 44 6D 6D 11 64 A4 0D  .3..C.'.qDmm.d..
01C0: 1A B2 BE 42 DA 68 02 2A   81 53 AD C2 B4 C9 05 11  ...B.h.*.S......
01D0: 58 6B 79 A6 1C 64 D7 2B   C9 0A AE 29 EB D2 C9 4C  Xky..d.+...)...L
01E0: D7 BE A3 04 6C AD 45 40   A9 EB 7D 39 3A 51 7E 27  ....l.E@...9:Q.'
01F0: 0C 9C AD BF 4A 90 58 86   5B 2F CC 5E F6 D8 6B 9A  ....J.X.[/.^..k.
0200: 9C C0 5C C9 C1 A2 07 35   AF E2 A3 D0 B5 E2 AA 10  ..\....5........
0210: 43 E2 40 46 DA 1C 38 7E   9F 6B 09 0B D2 91 ED 90  C.@F..8..k......
0220: 77 B8 D0 15 18 37 00 F6   4F AF 74 EA 78 F3 F6 84  w....7..O.t.x...
0230: BA 84 A0 F5 4F 37 83 14   F2 FA 08 01 9F 1A 20 F7  ....O7........ .
0240: 6B AC F4 F5 BD 76 73 E2   1A CA 91 5B AB 6D BE 81  k....vs....[.m..
0250: 9F 4A F1 5F 8E                                     .J._.

Entered Krb5Context.initSecContext with state=STATE_IN_PROCESS
Krb5Context setting peerSeqNumber to: 818076014
Debug is  true storeKey false useTicketCache false useKeyTab true doNotPrompt false ticketCache is null isInitiator true KeyTab is /Users/gaborsomogyi/docker-kerberos/mariadb_share/mariadb.keytab refreshKrb5Config is false principal is mariadb/example.com@EXAMPLE.COM tryFirstPass is false useFirstPass is false storePass is false clearPass is false
principal is mariadb/example.com@EXAMPLE.COM
Will use keytab
Commit Succeeded 

Found ticket for mariadb/example.com@EXAMPLE.COM to go to krbtgt/EXAMPLE.COM@EXAMPLE.COM expiring on Wed May 06 17:57:42 CEST 2020
Entered Krb5Context.initSecContext with state=STATE_NEW
Found ticket for mariadb/example.com@EXAMPLE.COM to go to krbtgt/EXAMPLE.COM@EXAMPLE.COM expiring on Wed May 06 17:57:42 CEST 2020
Service ticket not found in the subject
Krb5Context setting mySeqNumber to: 224955138
Created InitSecContextToken:
0000: 01 00 6E 82 02 4C 30 82   02 48 A0 03 02 01 05 A1  ..n..L0..H......
0010: 03 02 01 0E A2 07 03 05   00 20 00 00 00 A3 82 01  ......... ......
0020: 51 61 82 01 4D 30 82 01   49 A0 03 02 01 05 A1 0D  Qa..M0..I.......
0030: 1B 0B 45 58 41 4D 50 4C   45 2E 43 4F 4D A2 21 30  ..EXAMPLE.COM.!0
0040: 1F A0 03 02 01 03 A1 18   30 16 1B 07 6D 61 72 69  ........0...mari
0050: 61 64 62 1B 0B 65 78 61   6D 70 6C 65 2E 63 6F 6D  adb..example.com
0060: A3 82 01 0E 30 82 01 0A   A0 03 02 01 12 A1 03 02  ....0...........
0070: 01 02 A2 81 FD 04 81 FA   DE D0 E0 C9 4C F1 5B 52  ............L.[R
0080: 9E 33 F4 7D 06 42 63 7C   9C FC D6 CF A5 C7 83 FE  .3...Bc.........
0090: 4E D3 88 66 3D D8 D3 22   2A 8E 9A 3E A9 57 56 D2  N..f=.."*..>.WV.
00A0: 9F 63 A3 9C C5 30 3D 79   AB BC 00 41 80 2C ED 8A  .c...0=y...A.,..
00B0: AE B2 45 94 46 45 C8 DB   AB F8 78 58 D4 28 76 51  ..E.FE....xX.(vQ
00C0: 5E C7 EB DD 94 A5 12 98   9B BC DA 0A F2 E2 A2 A1  ^...............
00D0: 65 6A 5A 1E 7C C4 7E 06   E1 39 07 12 FD 1F 03 1D  ejZ......9......
00E0: 15 EF EB E1 FD 11 92 7B   F3 94 77 22 CC 07 A0 94  ..........w"....
00F0: D8 2B 1A 79 3E F9 0E 80   B0 66 D2 A2 3A 39 65 04  .+.y>....f..:9e.
0100: 58 13 F1 CF 66 01 76 DB   30 0E 07 1D AC 09 0C 6E  X...f.v.0......n
0110: 87 73 1C A2 B0 71 1E 9D   13 69 D3 71 D3 37 46 FF  .s...q...i.q.7F.
0120: EF E0 E7 0F DF F6 36 25   8B A3 49 59 BE 2C BC 4A  ......6%..IY.,.J
0130: 11 12 BA 0E C9 FA 0F 2C   41 69 2A 01 4F 5D 3D FB  .......,Ai*.O]=.
0140: 4D E0 3E 18 F5 05 20 72   B7 32 00 9F B9 A0 E8 26  M.>... r.2.....&
0150: 6B DD 4C 27 47 37 DD 3F   4C ED 34 26 59 B1 C8 5B  k.L'G7.?L.4&Y..[
0160: 56 D9 DE DA 1E 04 E0 DD   82 0E EA DB C0 12 6D C1  V.............m.
0170: 5B 01 A4 81 DD 30 81 DA   A0 03 02 01 12 A2 81 D2  [....0..........
0180: 04 81 CF C0 2E 7E 57 F2   E1 F9 0F BB 6F AC 9C 00  ......W.....o...
0190: 76 15 9B DE 37 3A B1 E6   94 99 28 CE 04 D9 3C 53  v...7:....(...<S
01A0: 19 61 84 CB 33 79 62 CA   EB F9 1F 22 7E 67 B6 A7  .a..3yb....".g..
01B0: E0 C3 A4 8C E9 BF 39 A3   8C 73 20 5B 91 35 DE A9  ......9..s [.5..
01C0: 67 45 E0 FC 61 FF AB EC   98 C6 C7 21 F8 75 5B CC  gE..a......!.u[.
01D0: 0F 7F BD 17 62 47 F7 2E   DD A2 59 1B F0 FE B2 9A  ....bG....Y.....
01E0: 93 86 97 2E E7 2D 49 E1   7D 3B D0 53 60 82 8F 67  .....-I..;.S`..g
01F0: F5 15 83 DF 83 5E 9F 47   80 DA 99 E8 4A 7C B5 7C  .....^.G....J...
0200: 66 EA 2F FD F4 41 C9 39   12 4B 49 CA 26 B4 0F 10  f./..A.9.KI.&...
0210: 87 AD 9E 47 8E 65 71 84   3B 41 F5 5D 58 B5 E4 91  ...G.eq.;A.]X...
0220: FD 31 4D 10 91 6B 79 7D   02 F0 45 6F 19 FF 0F F5  .1M..ky...Eo....
0230: B3 F0 AD 41 2F 56 80 0F   2A E1 32 80 EB 10 44 B8  ...A/V..*.2...D.
0240: 26 D6 9B 20 AC C0 B2 22   90 F9 52 15 CA 97 80 C6  &.. ..."..R.....
0250: 92 F4                                              ..

Entered Krb5Context.initSecContext with state=STATE_IN_PROCESS
Krb5Context setting peerSeqNumber to: 832652697
20/05/05 17:57:42.921 ScalaTest-run-running-MultiKrbIntegrationSuite INFO SharedState: loading hive config file: file:/Users/gaborsomogyi/spark/sql/core/target/scala-2.12/test-classes/hive-site.xml
20/05/05 17:57:43.009 ScalaTest-run-running-MultiKrbIntegrationSuite INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir ('file:/Users/gaborsomogyi/spark/external/docker-integration-tests/spark-warehouse//org.apache.spark.sql.jdbc.MultiKrbIntegrationSuite').
20/05/05 17:57:43.010 ScalaTest-run-running-MultiKrbIntegrationSuite INFO SharedState: Warehouse path is 'file:/Users/gaborsomogyi/spark/external/docker-integration-tests/spark-warehouse//org.apache.spark.sql.jdbc.MultiKrbIntegrationSuite'.
20/05/05 17:57:43.025 ScalaTest-run-running-MultiKrbIntegrationSuite INFO ContextHandler: Started o.e.j.s.ServletContextHandler@78b41097{/SQL,null,AVAILABLE,@Spark}
20/05/05 17:57:43.026 ScalaTest-run-running-MultiKrbIntegrationSuite INFO ContextHandler: Started o.e.j.s.ServletContextHandler@327c7bea{/SQL/json,null,AVAILABLE,@Spark}
20/05/05 17:57:43.027 ScalaTest-run-running-MultiKrbIntegrationSuite INFO ContextHandler: Started o.e.j.s.ServletContextHandler@20e6c4dc{/SQL/execution,null,AVAILABLE,@Spark}
20/05/05 17:57:43.027 ScalaTest-run-running-MultiKrbIntegrationSuite INFO ContextHandler: Started o.e.j.s.ServletContextHandler@4d2a1da3{/SQL/execution/json,null,AVAILABLE,@Spark}
20/05/05 17:57:43.029 ScalaTest-run-running-MultiKrbIntegrationSuite INFO ContextHandler: Started o.e.j.s.ServletContextHandler@557286ad{/static/sql,null,AVAILABLE,@Spark}
Debug is  true storeKey false useTicketCache false useKeyTab true doNotPrompt false ticketCache is null isInitiator true KeyTab is /Users/gaborsomogyi/docker-kerberos/postgres_share/postgres.keytab refreshKrb5Config is false principal is postgres/example.com@EXAMPLE.COM tryFirstPass is false useFirstPass is false storePass is false clearPass is false
principal is postgres/example.com@EXAMPLE.COM
Will use keytab
Commit Succeeded 

Found ticket for postgres/example.com@EXAMPLE.COM to go to krbtgt/EXAMPLE.COM@EXAMPLE.COM expiring on Wed May 06 17:57:43 CEST 2020
Entered Krb5Context.initSecContext with state=STATE_NEW
Found ticket for postgres/example.com@EXAMPLE.COM to go to krbtgt/EXAMPLE.COM@EXAMPLE.COM expiring on Wed May 06 17:57:43 CEST 2020
Service ticket not found in the subject
Krb5Context setting mySeqNumber to: 903124025
Created InitSecContextToken:
0000: 01 00 6E 82 02 4F 30 82   02 4B A0 03 02 01 05 A1  ..n..O0..K......
0010: 03 02 01 0E A2 07 03 05   00 20 00 00 00 A3 82 01  ......... ......
0020: 53 61 82 01 4F 30 82 01   4B A0 03 02 01 05 A1 0D  Sa..O0..K.......
0030: 1B 0B 45 58 41 4D 50 4C   45 2E 43 4F 4D A2 22 30  ..EXAMPLE.COM."0
0040: 20 A0 03 02 01 03 A1 19   30 17 1B 08 70 6F 73 74   .......0...post
0050: 67 72 65 73 1B 0B 65 78   61 6D 70 6C 65 2E 63 6F  gres..example.co
0060: 6D A3 82 01 0F 30 82 01   0B A0 03 02 01 12 A1 03  m....0..........
0070: 02 01 02 A2 81 FE 04 81   FB A6 36 55 18 AF 7F 4A  ..........6U...J
0080: 88 77 49 1F 68 66 E0 C9   85 58 14 A7 F8 74 E5 B5  .wI.hf...X...t..
0090: 6F A0 2A 2C CF 0F 0C C9   B5 A5 5E A1 AE D8 CB BA  o.*,......^.....
00A0: FF 44 6E 45 05 56 A3 75   83 3C 63 A2 FF F0 B6 4F  .DnE.V.u.<c....O
00B0: 7C 4E D6 BD 88 3F 22 54   2E 83 F9 F0 6A B4 6A 4B  .N...?"T....j.jK
00C0: BC 29 97 93 32 69 94 03   DF 32 C9 68 2C E2 B5 0A  .)..2i...2.h,...
00D0: D1 F5 53 16 03 E1 6A 6D   FC 35 A9 49 5B 6D 96 FE  ..S...jm.5.I[m..
00E0: B6 58 93 BE 25 D1 DE 71   7D 61 2A 96 8B 0A 14 45  .X..%..q.a*....E
00F0: 51 D0 7C 3A 28 DB 06 12   28 77 24 3D 78 4C 5C 41  Q..:(...(w$=xL\A
0100: 48 0C 56 00 BD DB 1C 94   57 4C B9 F4 91 86 37 FF  H.V.....WL....7.
0110: 47 01 6B 73 AD BD 5A 4C   BA 39 BE BF 1A 59 D9 EB  G.ks..ZL.9...Y..
0120: 41 6D 62 EA EE F4 55 C8   DC C2 2E 84 7C 5B 58 24  Amb...U......[X$
0130: A1 83 76 E6 F5 68 46 35   27 8A ED 8F FC AA A1 89  ..v..hF5'.......
0140: CD DF A1 AC 20 F1 AD 29   ED 96 80 9B 26 A2 1A 5C  .... ..)....&..\
0150: AD 5E B0 02 A2 38 0F DE   40 27 14 CF 00 94 35 BD  .^...8..@'....5.
0160: 52 96 48 3D 4D 3C CC 9C   7A CC 82 C3 4C BC B1 60  R.H=M<..z...L..`
0170: 5B BB DF 0E A4 81 DE 30   81 DB A0 03 02 01 12 A2  [......0........
0180: 81 D3 04 81 D0 CC B5 76   2E 75 66 39 5E 40 EF CB  .......v.uf9^@..
0190: 52 41 74 C0 4C 96 88 3B   35 32 FF CD 95 B1 4E 49  RAt.L..;52....NI
01A0: 2A 47 9A 7E 3F A1 97 67   7C D4 47 C4 B9 C6 77 D4  *G..?..g..G...w.
01B0: BD E1 98 C3 FD 4E 2E 36   B0 70 54 F3 46 D5 2A FB  .....N.6.pT.F.*.
01C0: FD CE AE 4B 36 F6 14 09   12 38 BB 79 D2 15 A9 33  ...K6....8.y...3
01D0: E6 D7 12 9D 26 27 64 E7   C7 41 F0 C2 16 7C F1 97  ....&'d..A......
01E0: F0 63 C3 6B A4 E7 FA 86   7F F9 15 4D AA 14 2C F5  .c.k.......M..,.
01F0: A9 FD DA 26 27 01 6D 0D   4C 20 7D 4E F3 93 1D 88  ...&'.m.L .N....
0200: B5 86 77 39 08 DF C7 E0   16 6E 78 8D 25 BC DF 7C  ..w9.....nx.%...
0210: B6 93 4F C2 E4 C6 A4 69   2F 77 58 23 2B 3B 19 49  ..O....i/wX#+;.I
0220: 5E C3 B1 CC 5A 89 91 35   B7 A4 09 EE 51 82 1C E2  ^...Z..5....Q...
0230: 66 F8 29 3A EF F8 C5 95   3A D6 39 EC F2 F6 76 BB  f.):....:.9...v.
0240: 8F 8E A3 AF 99 81 EF 7D   18 23 71 A0 DC DF FF 0F  .........#q.....
0250: 79 DA 1F 65 0D                                     y..e.

Entered Krb5Context.initSecContext with state=STATE_IN_PROCESS
Krb5Context setting peerSeqNumber to: 929779812
20/05/05 17:57:45.413 ScalaTest-run-running-MultiKrbIntegrationSuite INFO BaseSessionStateBuilder$$anon$2: Optimization rule 'org.apache.spark.sql.catalyst.optimizer.ConvertToLocalRelation' is excluded from the optimizer.
20/05/05 17:57:45.413 ScalaTest-run-running-MultiKrbIntegrationSuite INFO BaseSessionStateBuilder$$anon$2: Optimization rule 'org.apache.spark.sql.catalyst.optimizer.ConvertToLocalRelation' is excluded from the optimizer.
20/05/05 17:57:46.215 ScalaTest-run-running-MultiKrbIntegrationSuite INFO CodeGenerator: Code generated in 198.742307 ms
20/05/05 17:57:46.344 ScalaTest-run-running-MultiKrbIntegrationSuite INFO SparkContext: Starting job: $anonfun$new$1 at OutcomeOf.scala:85
20/05/05 17:57:46.361 dag-scheduler-event-loop INFO DAGScheduler: Got job 0 ($anonfun$new$1 at OutcomeOf.scala:85) with 1 output partitions
20/05/05 17:57:46.361 dag-scheduler-event-loop INFO DAGScheduler: Final stage: ResultStage 0 ($anonfun$new$1 at OutcomeOf.scala:85)
20/05/05 17:57:46.361 dag-scheduler-event-loop INFO DAGScheduler: Parents of final stage: List()
20/05/05 17:57:46.362 dag-scheduler-event-loop INFO DAGScheduler: Missing parents: List()
20/05/05 17:57:46.367 dag-scheduler-event-loop INFO DAGScheduler: Submitting ResultStage 0 (MapPartitionsRDD[2] at $anonfun$new$1 at OutcomeOf.scala:85), which has no missing parents
20/05/05 17:57:46.438 dag-scheduler-event-loop INFO MemoryStore: Block broadcast_0 stored as values in memory (estimated size 8.8 KiB, free 4.1 GiB)
20/05/05 17:57:46.484 dag-scheduler-event-loop INFO MemoryStore: Block broadcast_0_piece0 stored as bytes in memory (estimated size 4.7 KiB, free 4.1 GiB)
20/05/05 17:57:46.488 dispatcher-BlockManagerMaster INFO BlockManagerInfo: Added broadcast_0_piece0 in memory on example.com:51449 (size: 4.7 KiB, free: 4.1 GiB)
20/05/05 17:57:46.491 dag-scheduler-event-loop INFO SparkContext: Created broadcast 0 from broadcast at DAGScheduler.scala:1293
20/05/05 17:57:46.510 dag-scheduler-event-loop INFO DAGScheduler: Submitting 1 missing tasks from ResultStage 0 (MapPartitionsRDD[2] at $anonfun$new$1 at OutcomeOf.scala:85) (first 15 tasks are for partitions Vector(0))
20/05/05 17:57:46.512 dag-scheduler-event-loop INFO TaskSchedulerImpl: Adding task set 0.0 with 1 tasks resource profile 0
20/05/05 17:57:46.573 dispatcher-event-loop-0 INFO TaskSetManager: Starting task 0.0 in stage 0.0 (TID 0, example.com, executor driver, partition 0, PROCESS_LOCAL, 7171 bytes) taskResourceAssignments Map()
20/05/05 17:57:46.586 Executor task launch worker for task 0 INFO Executor: Running task 0.0 in stage 0.0 (TID 0)
Debug is  true storeKey false useTicketCache false useKeyTab true doNotPrompt false ticketCache is null isInitiator true KeyTab is /Users/gaborsomogyi/docker-kerberos/postgres_share/postgres.keytab refreshKrb5Config is false principal is postgres/example.com@EXAMPLE.COM tryFirstPass is false useFirstPass is false storePass is false clearPass is false
principal is postgres/example.com@EXAMPLE.COM
Will use keytab
Commit Succeeded 

Found ticket for postgres/example.com@EXAMPLE.COM to go to krbtgt/EXAMPLE.COM@EXAMPLE.COM expiring on Wed May 06 17:57:46 CEST 2020
Entered Krb5Context.initSecContext with state=STATE_NEW
Found ticket for postgres/example.com@EXAMPLE.COM to go to krbtgt/EXAMPLE.COM@EXAMPLE.COM expiring on Wed May 06 17:57:46 CEST 2020
Service ticket not found in the subject
Krb5Context setting mySeqNumber to: 356384901
Created InitSecContextToken:
0000: 01 00 6E 82 02 4F 30 82   02 4B A0 03 02 01 05 A1  ..n..O0..K......
0010: 03 02 01 0E A2 07 03 05   00 20 00 00 00 A3 82 01  ......... ......
0020: 53 61 82 01 4F 30 82 01   4B A0 03 02 01 05 A1 0D  Sa..O0..K.......
0030: 1B 0B 45 58 41 4D 50 4C   45 2E 43 4F 4D A2 22 30  ..EXAMPLE.COM."0
0040: 20 A0 03 02 01 03 A1 19   30 17 1B 08 70 6F 73 74   .......0...post
0050: 67 72 65 73 1B 0B 65 78   61 6D 70 6C 65 2E 63 6F  gres..example.co
0060: 6D A3 82 01 0F 30 82 01   0B A0 03 02 01 12 A1 03  m....0..........
0070: 02 01 02 A2 81 FE 04 81   FB 08 4E 13 18 6A B0 D4  ..........N..j..
0080: 38 C1 C8 BD 57 EF 82 19   24 FC 5D 2B 3B 63 48 BE  8...W...$.]+;cH.
0090: DF D6 24 1D 74 BB 51 7B   D1 94 58 AE E9 FF D9 45  ..$.t.Q...X....E
00A0: 14 8B 1D AE E0 9F 2B 2F   FF 71 9E 93 84 5A 38 EF  ......+/.q...Z8.
00B0: 95 59 B2 04 1E BE 0C D7   E6 A5 C7 C6 F5 A1 2E E8  .Y..............
00C0: 45 C9 E7 19 3C 9D 95 E3   46 6A 8A 7D B3 1E 2F 8D  E...<...Fj..../.
00D0: 40 DA AD 01 E1 5E D2 2B   47 3E 24 52 66 83 9A 0D  @....^.+G>$Rf...
00E0: 75 A4 6E DA F5 A3 58 E1   5A 34 27 2F 5E 1D A7 0D  u.n...X.Z4'/^...
00F0: 16 E4 1C F2 D6 28 88 08   BF 06 07 25 CA A7 0E 28  .....(.....%...(
0100: CD D1 47 4A 01 CB 8E 44   0C A5 30 5D 2C BA 57 19  ..GJ...D..0],.W.
0110: 77 E0 4C 40 9F 2F 9D 13   C2 FB AB F6 0C 17 B8 EB  w.L@./..........
0120: FB 34 6B FC C5 1D 23 14   89 D9 A6 A6 C7 A1 B6 2C  .4k...#........,
0130: 72 C5 61 50 B8 4E 58 77   19 40 E0 D4 2D 9D 53 FA  r.aP.NXw.@..-.S.
0140: B4 6D B1 BB 07 02 8D 5B   5B 00 B0 11 CD 28 79 75  .m.....[[....(yu
0150: 12 4E 5D 04 A8 88 4A F7   FD 31 4F 79 0F 6D CE 79  .N]...J..1Oy.m.y
0160: 3A C0 51 59 EB AC A9 AE   00 E4 D2 7A E4 E5 9C A1  :.QY.......z....
0170: 32 63 0E CA A4 81 DE 30   81 DB A0 03 02 01 12 A2  2c.....0........
0180: 81 D3 04 81 D0 25 DD CF   96 A5 5B C6 46 5A BC 47  .....%....[.FZ.G
0190: 03 63 FF 79 74 39 78 E0   93 C2 7D 61 D7 7D 34 83  .c.yt9x....a..4.
01A0: 65 85 1B 3B 22 38 DB 58   A4 A8 6B 3B 3C 9F 15 C1  e..;"8.X..k;<...
01B0: D9 1B 61 BD C2 87 5F D3   E8 90 A1 F3 04 04 AA AC  ..a..._.........
01C0: 9E FF 3A 17 FF AB 69 2D   DC 05 CB F4 31 5D 80 EC  ..:...i-....1]..
01D0: 33 0A 00 D8 D1 C2 B8 D5   D6 7E A8 09 E1 67 E5 5A  3............g.Z
01E0: 4D A0 8A 58 17 EE DE F4   35 C0 8E C2 CD AC 39 B5  M..X....5.....9.
01F0: DD D7 04 08 41 7F E1 6E   E9 98 D4 FC CF 14 4D 5D  ....A..n......M]
0200: 98 C6 86 FF 50 A5 C5 9D   3D 95 77 24 85 C9 6A 99  ....P...=.w$..j.
0210: 73 52 85 D0 2B 92 2C 0F   AB 43 7E AF 87 0E 16 C5  sR..+.,..C......
0220: 1F 19 36 00 1C CC 0D 5A   92 B3 08 B5 92 72 41 DA  ..6....Z.....rA.
0230: 1F F0 54 46 F7 83 11 44   A8 B3 89 3A 74 3C 5B 8F  ..TF...D...:t<[.
0240: 76 B1 7A D2 C5 1F 1D 26   F8 DC 51 C2 BD 97 03 05  v.z....&..Q.....
0250: B0 78 CF 74 B9                                     .x.t.

Entered Krb5Context.initSecContext with state=STATE_IN_PROCESS
Krb5Context setting peerSeqNumber to: 593736185
20/05/05 17:57:46.921 Executor task launch worker for task 0 INFO JDBCRDD: closed connection
20/05/05 17:57:46.937 Executor task launch worker for task 0 INFO Executor: Finished task 0.0 in stage 0.0 (TID 0). 1498 bytes result sent to driver
20/05/05 17:57:47.048 task-result-getter-0 INFO TaskSetManager: Finished task 0.0 in stage 0.0 (TID 0) in 376 ms on example.com (executor driver) (1/1)
20/05/05 17:57:47.050 task-result-getter-0 INFO TaskSchedulerImpl: Removed TaskSet 0.0, whose tasks have all completed, from pool 
20/05/05 17:57:47.058 dag-scheduler-event-loop INFO DAGScheduler: ResultStage 0 ($anonfun$new$1 at OutcomeOf.scala:85) finished in 0.672 s
20/05/05 17:57:47.061 dag-scheduler-event-loop INFO DAGScheduler: Job 0 is finished. Cancelling potential speculative or zombie tasks for this job
20/05/05 17:57:47.061 dag-scheduler-event-loop INFO TaskSchedulerImpl: Killing all running tasks in stage 0: Stage finished
20/05/05 17:57:47.063 ScalaTest-run-running-MultiKrbIntegrationSuite INFO DAGScheduler: Job 0 finished: $anonfun$new$1 at OutcomeOf.scala:85, took 0.718061 s
20/05/05 17:57:47.095 ScalaTest-run-running-MultiKrbIntegrationSuite INFO CodeGenerator: Code generated in 16.107 ms
Debug is  true storeKey false useTicketCache false useKeyTab true doNotPrompt false ticketCache is null isInitiator true KeyTab is /Users/gaborsomogyi/docker-kerberos/mariadb_share/mariadb.keytab refreshKrb5Config is false principal is mariadb/example.com@EXAMPLE.COM tryFirstPass is false useFirstPass is false storePass is false clearPass is false
principal is mariadb/example.com@EXAMPLE.COM
Will use keytab
Commit Succeeded 

Found ticket for mariadb/example.com@EXAMPLE.COM to go to krbtgt/EXAMPLE.COM@EXAMPLE.COM expiring on Wed May 06 17:57:47 CEST 2020
Entered Krb5Context.initSecContext with state=STATE_NEW
Found ticket for mariadb/example.com@EXAMPLE.COM to go to krbtgt/EXAMPLE.COM@EXAMPLE.COM expiring on Wed May 06 17:57:47 CEST 2020
Service ticket not found in the subject
Krb5Context setting mySeqNumber to: 881086553
Created InitSecContextToken:
0000: 01 00 6E 82 02 4C 30 82   02 48 A0 03 02 01 05 A1  ..n..L0..H......
0010: 03 02 01 0E A2 07 03 05   00 20 00 00 00 A3 82 01  ......... ......
0020: 51 61 82 01 4D 30 82 01   49 A0 03 02 01 05 A1 0D  Qa..M0..I.......
0030: 1B 0B 45 58 41 4D 50 4C   45 2E 43 4F 4D A2 21 30  ..EXAMPLE.COM.!0
0040: 1F A0 03 02 01 03 A1 18   30 16 1B 07 6D 61 72 69  ........0...mari
0050: 61 64 62 1B 0B 65 78 61   6D 70 6C 65 2E 63 6F 6D  adb..example.com
0060: A3 82 01 0E 30 82 01 0A   A0 03 02 01 12 A1 03 02  ....0...........
0070: 01 02 A2 81 FD 04 81 FA   03 0A 90 33 C8 E4 70 FE  ...........3..p.
0080: 0E 27 C9 9F 4F 7E 00 68   FC 26 79 B3 4B 79 3C 40  .'..O..h.&y.Ky<@
0090: 63 11 A3 1B 02 97 42 F5   68 C4 1A 74 4B 21 92 04  c.....B.h..tK!..
00A0: 63 D7 9B 00 A3 7F D0 6A   42 7A D6 D9 91 F8 6D 15  c......jBz....m.
00B0: 1B 9F AA 15 55 D8 A3 2A   10 86 D9 F1 47 25 07 41  ....U..*....G%.A
00C0: 00 FA C6 49 C2 51 C6 23   91 DE B0 1E 37 A1 25 06  ...I.Q.#....7.%.
00D0: A9 60 63 09 E2 19 30 F5   8E C8 27 13 67 B6 E8 6D  .`c...0...'.g..m
00E0: 08 87 98 4F B2 F3 8F 5E   F5 E6 79 ED B8 4F 3E A7  ...O...^..y..O>.
00F0: D5 AF 56 87 5C 37 2F 73   68 AD 5D C4 01 5D 99 D1  ..V.\7/sh.]..]..
0100: 1F F7 B9 9D D8 C9 DC 0A   51 3B 33 4C E1 76 E3 32  ........Q;3L.v.2
0110: A7 D4 FE 05 4F 92 B2 04   CF 2B AC 21 D2 81 C9 10  ....O....+.!....
0120: 09 BB 36 BE EB 7D 8A 7E   AA AE AB 11 48 A9 08 39  ..6.........H..9
0130: 14 F2 8D EF 54 EF E3 18   48 C6 77 AF 28 01 4D 52  ....T...H.w.(.MR
0140: DC 68 76 4F 48 E3 71 34   73 0C 35 46 9B BE 9A 34  .hvOH.q4s.5F...4
0150: E0 0B AF 65 58 E3 91 DC   01 25 82 99 45 77 9D 71  ...eX....%..Ew.q
0160: C6 C5 49 CC 10 46 63 53   FC 16 ED 1A 9D EA 93 1E  ..I..FcS........
0170: CD 71 A4 81 DD 30 81 DA   A0 03 02 01 12 A2 81 D2  .q...0..........
0180: 04 81 CF 32 DF C6 B4 1C   9E 53 4D 20 B3 78 66 90  ...2.....SM .xf.
0190: 38 FB 8B F8 A7 6A 59 86   EC A8 AC 13 38 90 BC 81  8....jY.....8...
01A0: E8 FF 69 4B 05 4B 0A 87   45 ED 8D FE CB EB 97 BA  ..iK.K..E.......
01B0: 67 51 E4 77 C7 D0 13 95   CA B5 64 92 D8 28 0A C8  gQ.w......d..(..
01C0: 9A FF F5 64 41 E5 5C 2B   3C 28 05 8A 65 B4 05 2F  ...dA.\+<(..e../
01D0: 4E C4 FD 82 08 30 AD 1C   8A 35 76 02 AC B7 83 96  N....0...5v.....
01E0: E0 E5 A9 46 B4 B5 43 0C   C9 FE 27 B3 BA E0 46 B2  ...F..C...'...F.
01F0: 5A 4B BF 67 3A 3E 16 66   3E 71 7B 07 6E 7C F2 AA  ZK.g:>.f>q..n...
0200: 77 65 CA 59 4D FA A1 93   C3 20 83 91 0F 56 67 19  we.YM.... ...Vg.
0210: C3 BC 47 CF 96 C8 67 EF   BA 5F 68 A5 1C D4 B5 C9  ..G...g.._h.....
0220: 2D 9C BE 8B DE B5 61 6D   DE AE 9E FC ED 16 86 69  -.....am.......i
0230: 60 2D 6B 40 B9 D4 6A C3   43 84 B1 58 6E 2C F5 FC  `-k@..j.C..Xn,..
0240: 0D 56 17 7E 2B 1D C6 F8   DF 8C 0A 65 39 9D AF 5D  .V..+......e9..]
0250: 5B 50                                              [P

Entered Krb5Context.initSecContext with state=STATE_IN_PROCESS
Krb5Context setting peerSeqNumber to: 279045794
20/05/05 17:57:47.165 ScalaTest-run-running-MultiKrbIntegrationSuite INFO BaseSessionStateBuilder$$anon$2: Optimization rule 'org.apache.spark.sql.catalyst.optimizer.ConvertToLocalRelation' is excluded from the optimizer.
20/05/05 17:57:47.166 ScalaTest-run-running-MultiKrbIntegrationSuite INFO BaseSessionStateBuilder$$anon$2: Optimization rule 'org.apache.spark.sql.catalyst.optimizer.ConvertToLocalRelation' is excluded from the optimizer.
20/05/05 17:57:47.215 ScalaTest-run-running-MultiKrbIntegrationSuite INFO SparkContext: Starting job: $anonfun$new$1 at OutcomeOf.scala:85
20/05/05 17:57:47.216 dag-scheduler-event-loop INFO DAGScheduler: Got job 1 ($anonfun$new$1 at OutcomeOf.scala:85) with 1 output partitions
20/05/05 17:57:47.216 dag-scheduler-event-loop INFO DAGScheduler: Final stage: ResultStage 1 ($anonfun$new$1 at OutcomeOf.scala:85)
20/05/05 17:57:47.216 dag-scheduler-event-loop INFO DAGScheduler: Parents of final stage: List()
20/05/05 17:57:47.217 dag-scheduler-event-loop INFO DAGScheduler: Missing parents: List()
20/05/05 17:57:47.218 dag-scheduler-event-loop INFO DAGScheduler: Submitting ResultStage 1 (MapPartitionsRDD[5] at $anonfun$new$1 at OutcomeOf.scala:85), which has no missing parents
20/05/05 17:57:47.223 dag-scheduler-event-loop INFO MemoryStore: Block broadcast_1 stored as values in memory (estimated size 8.8 KiB, free 4.1 GiB)
20/05/05 17:57:47.224 dag-scheduler-event-loop INFO MemoryStore: Block broadcast_1_piece0 stored as bytes in memory (estimated size 4.6 KiB, free 4.1 GiB)
20/05/05 17:57:47.225 dispatcher-BlockManagerMaster INFO BlockManagerInfo: Added broadcast_1_piece0 in memory on example.com:51449 (size: 4.6 KiB, free: 4.1 GiB)
20/05/05 17:57:47.226 dag-scheduler-event-loop INFO SparkContext: Created broadcast 1 from broadcast at DAGScheduler.scala:1293
20/05/05 17:57:47.227 dag-scheduler-event-loop INFO DAGScheduler: Submitting 1 missing tasks from ResultStage 1 (MapPartitionsRDD[5] at $anonfun$new$1 at OutcomeOf.scala:85) (first 15 tasks are for partitions Vector(0))
20/05/05 17:57:47.227 dag-scheduler-event-loop INFO TaskSchedulerImpl: Adding task set 1.0 with 1 tasks resource profile 0
20/05/05 17:57:47.228 dispatcher-event-loop-1 INFO TaskSetManager: Starting task 0.0 in stage 1.0 (TID 1, example.com, executor driver, partition 0, PROCESS_LOCAL, 7171 bytes) taskResourceAssignments Map()
20/05/05 17:57:47.229 Executor task launch worker for task 1 INFO Executor: Running task 0.0 in stage 1.0 (TID 1)
Debug is  true storeKey false useTicketCache false useKeyTab true doNotPrompt false ticketCache is null isInitiator true KeyTab is /Users/gaborsomogyi/docker-kerberos/mariadb_share/mariadb.keytab refreshKrb5Config is false principal is mariadb/example.com@EXAMPLE.COM tryFirstPass is false useFirstPass is false storePass is false clearPass is false
principal is mariadb/example.com@EXAMPLE.COM
Will use keytab
Commit Succeeded 

Found ticket for mariadb/example.com@EXAMPLE.COM to go to krbtgt/EXAMPLE.COM@EXAMPLE.COM expiring on Wed May 06 17:57:47 CEST 2020
Entered Krb5Context.initSecContext with state=STATE_NEW
Found ticket for mariadb/example.com@EXAMPLE.COM to go to krbtgt/EXAMPLE.COM@EXAMPLE.COM expiring on Wed May 06 17:57:47 CEST 2020
Service ticket not found in the subject
Krb5Context setting mySeqNumber to: 874492690
Created InitSecContextToken:
0000: 01 00 6E 82 02 4C 30 82   02 48 A0 03 02 01 05 A1  ..n..L0..H......
0010: 03 02 01 0E A2 07 03 05   00 20 00 00 00 A3 82 01  ......... ......
0020: 51 61 82 01 4D 30 82 01   49 A0 03 02 01 05 A1 0D  Qa..M0..I.......
0030: 1B 0B 45 58 41 4D 50 4C   45 2E 43 4F 4D A2 21 30  ..EXAMPLE.COM.!0
0040: 1F A0 03 02 01 03 A1 18   30 16 1B 07 6D 61 72 69  ........0...mari
0050: 61 64 62 1B 0B 65 78 61   6D 70 6C 65 2E 63 6F 6D  adb..example.com
0060: A3 82 01 0E 30 82 01 0A   A0 03 02 01 12 A1 03 02  ....0...........
0070: 01 02 A2 81 FD 04 81 FA   CF 2C 16 7B F6 BE BE D0  .........,......
0080: 43 92 81 20 46 84 C2 2B   65 CC 83 0E 81 95 56 3D  C.. F..+e.....V=
0090: 80 57 97 1D 67 CC F1 16   D0 40 17 BC D2 7B 22 4E  .W..g....@...."N
00A0: 71 DA A3 08 26 DF 31 11   73 B2 7C 05 A6 02 95 CB  q...&.1.s.......
00B0: C6 CE 72 6C 52 C4 0D C7   9F 80 06 E7 FA A6 DC F6  ..rlR...........
00C0: 0A B6 C3 9A CF 43 9D A8   3D 31 4B 4B 5B 8A 40 5B  .....C..=1KK[.@[
00D0: 6B F3 96 DE FB F3 32 7F   8D 94 E2 9E 32 C2 5A 8C  k.....2.....2.Z.
00E0: 4B 0D 1A EF 99 FD 73 BF   7D 11 69 CF E9 DA 01 89  K.....s...i.....
00F0: 17 12 15 30 0E 0C 39 7E   16 75 0E 99 00 BB DC AD  ...0..9..u......
0100: 05 00 AD FB 69 BB A5 B4   05 BD 15 C0 C4 67 AE 1A  ....i........g..
0110: 17 4A BE 3B C4 39 66 28   31 02 33 FF 8F B7 48 21  .J.;.9f(1.3...H!
0120: C8 F3 0F 56 76 F7 1E 86   FE 25 D8 8B 72 B0 17 11  ...Vv....%..r...
0130: 71 60 F0 18 FB 9A DA CD   99 98 53 49 FB 44 AB 95  q`........SI.D..
0140: B1 1A 50 85 F6 9C 44 37   F0 B3 F2 C6 52 AB EE D7  ..P...D7....R...
0150: D5 1C 5D 19 6A 31 B6 98   48 DD 49 07 8D 4B D0 71  ..].j1..H.I..K.q
0160: 52 F4 C1 35 A1 51 B5 9D   B8 0C BF 05 CA 85 03 51  R..5.Q.........Q
0170: 36 68 A4 81 DD 30 81 DA   A0 03 02 01 12 A2 81 D2  6h...0..........
0180: 04 81 CF 27 FA BE AA 98   90 43 3B A8 6D 06 63 0D  ...'.....C;.m.c.
0190: 23 0A 96 87 96 61 8B 05   96 DF 06 45 24 A4 22 96  #....a.....E$.".
01A0: 89 3D 74 2C 97 FC 26 CD   DD 93 45 BB 3F F2 46 2E  .=t,..&...E.?.F.
01B0: 6A 33 9D BD 17 B2 62 F8   E8 80 05 9B 45 78 EE 03  j3....b.....Ex..
01C0: 93 5C 2F C9 7C 9D E2 94   35 13 E1 C5 CF C1 83 24  .\/.....5......$
01D0: DF CD E7 64 6C 8B 36 2D   C3 E7 7E CB 69 9D 0E 15  ...dl.6-....i...
01E0: FD 05 C9 61 DB 9D 66 68   81 39 E6 50 41 7D A0 2C  ...a..fh.9.PA..,
01F0: A5 E1 89 45 3C 19 E1 49   E8 E2 52 A3 F9 46 07 D1  ...E<..I..R..F..
0200: C6 4D 1E 2B BC 3A 83 7D   7D 0F B3 1D 65 B9 43 09  .M.+.:......e.C.
0210: 61 1C 8C 86 AB 6B A3 41   56 1F CB 36 9B D6 66 BD  a....k.AV..6..f.
0220: 2C B4 AB 68 90 A4 BA 4B   8E 8E 49 AA 82 8E 50 66  ,..h...K..I...Pf
0230: DA 96 BD 2A 44 87 6E CA   A9 37 DF 89 E9 63 1B AB  ...*D.n..7...c..
0240: 2D 50 18 EB F6 86 7A 30   35 86 D9 BE 9C 46 31 0A  -P....z05....F1.
0250: D7 BA                                              ..

Entered Krb5Context.initSecContext with state=STATE_IN_PROCESS
Krb5Context setting peerSeqNumber to: 665129561
20/05/05 17:57:47.262 Executor task launch worker for task 1 INFO JDBCRDD: closed connection
20/05/05 17:57:47.264 Executor task launch worker for task 1 INFO Executor: Finished task 0.0 in stage 1.0 (TID 1). 1455 bytes result sent to driver
20/05/05 17:57:47.265 task-result-getter-1 INFO TaskSetManager: Finished task 0.0 in stage 1.0 (TID 1) in 37 ms on example.com (executor driver) (1/1)
20/05/05 17:57:47.265 task-result-getter-1 INFO TaskSchedulerImpl: Removed TaskSet 1.0, whose tasks have all completed, from pool 
20/05/05 17:57:47.265 dag-scheduler-event-loop INFO DAGScheduler: ResultStage 1 ($anonfun$new$1 at OutcomeOf.scala:85) finished in 0.046 s
20/05/05 17:57:47.266 dag-scheduler-event-loop INFO DAGScheduler: Job 1 is finished. Cancelling potential speculative or zombie tasks for this job
20/05/05 17:57:47.266 dag-scheduler-event-loop INFO TaskSchedulerImpl: Killing all running tasks in stage 1: Stage finished
20/05/05 17:57:47.266 ScalaTest-run-running-MultiKrbIntegrationSuite INFO DAGScheduler: Job 1 finished: $anonfun$new$1 at OutcomeOf.scala:85, took 0.050513 s
20/05/05 17:57:47.269 ScalaTest-run-running-MultiKrbIntegrationSuite INFO MultiKrbIntegrationSuite: 

===== FINISHED o.a.s.sql.jdbc.MultiKrbIntegrationSuite: 'Basic multi test' =====


20/05/05 17:57:47.717 ScalaTest-run INFO AbstractConnector: Stopped Spark@4985cbcb{HTTP/1.1,[http/1.1]}{0.0.0.0:4040}
20/05/05 17:57:47.719 ScalaTest-run INFO SparkUI: Stopped Spark web UI at http://example.com:4040
20/05/05 17:57:47.729 dispatcher-event-loop-0 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
20/05/05 17:57:47.741 ScalaTest-run INFO MemoryStore: MemoryStore cleared
20/05/05 17:57:47.742 ScalaTest-run INFO BlockManager: BlockManager stopped
20/05/05 17:57:47.748 ScalaTest-run INFO BlockManagerMaster: BlockManagerMaster stopped
20/05/05 17:57:47.750 dispatcher-event-loop-0 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
20/05/05 17:57:47.759 ScalaTest-run INFO SparkContext: Successfully stopped SparkContext
20/05/05 17:57:47.761 ScalaTest-run WARN MultiKrbIntegrationSuite: 

===== POSSIBLE THREAD LEAK IN SUITE o.a.s.sql.jdbc.MultiKrbIntegrationSuite, thread names: rpc-boss-3-1, shuffle-boss-6-1 =====


20/05/05 17:57:47.768 Thread-1 INFO ShutdownHookManager: Shutdown hook called
20/05/05 17:57:47.769 Thread-1 INFO ShutdownHookManager: Deleting directory /private/var/folders/jd/35_sh46s7zq0qc6khfw8hc800000gn/T/spark-81edaafe-d777-410a-96c3-42d26b880237

Process finished with exit code 0
```

Reply:
Of course this tests the fact that multiple database instances can be accessed from the same JVM and not the race itself.

Reply:
WDYT guys?

Reply:
Thanks for the detailed info, @gaborgsomogyi . btw, is there any reference (documents, specification, ...) to the behaviour you described above?
```
Worth to mention all the originally existing configuration is still reachable because if the searched application name is not matching with the newly added entry then the entry delegates the call to the parent.
```

Reply:
I haven't added any reference because:
* This pattern in security is a standard
* This is a detail from user perspective. The important thing from that perspective is that every previously configured entry is still visible, untouched and the configured database credentials will be configured internally.

What I can imagine is a similar internal doc for developers like [this](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/deploy/security/README.md).